File: Practicas/Juego Qytetet/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/titulo_propiedad.rb
#encoding: utf-8

module ModeloQytetet
  
  class TituloPropiedad
    
    attr_accessor :nombre #Strign
    attr_accessor :hipotecada #boleano
    attr_accessor :alquilerBase #int
    attr_accessor :factorRevalorizacion #float
    attr_accessor :hipotecaBase #int
    attr_accessor :precio_edificar #int
    
    #atributo clase jugador
    attr_accessor :propietario
    
    #atributo clase Casilla
    attr_accessor :casilla
    
    
    #---------------------------CONSTRUCTORES----------------------------
  
    def initialize(nombre, alquilerBase, factorRevalorizacion, hipotecaBase, precio_edificar)#,casilla);
      @nombre = nombre
      @alquilerBase = alquilerBase
      @factorRevalorizacion = factorRevalorizacion
      @hipotecada = false
      @hipotecaBase = hipotecaBase
      @precio_edificar = precio_edificar
      @propietario = nil
      @casilla = nil
    end

    #----------------------------------------------------------------------

    #------------------------------METODOS---------------------------------
    
    def cobrar_alquiler(coste_alquiler)
      @propietario.modificar_saldo(coste_alquiler)
    end
    
    def propietario_encarcelado
      @propietario.encarcelado
    end
    
    def tengo_propietario
      tengo = false
      if(@propietario != nil)
        tengo = true
      end
      tengo
    end
    
    
    #---------------------------TO_S-------------------------------------
    def to_s()
      "\n       Nombre: #{@nombre} 
       Hipotecada: #{@hipotecada} 
       Alquiler Base: #{@alquilerBase} 
       Factor Revalorizacion: #{@factorRevalorizacion} 
       Hipote Base: #{@hipotecaBase} 
       Precio Edificar: #{@precio_edificar}  
       Propietario: #{@propietario}
       Casilla: #{@casilla}
      "
    end
      
  end
end

File: Practicas/Juego Qytetet/QytetetJava/src/modeloqytetet/TituloPropiedad.java

package modeloqytetet;

/**
 *
 * @author albertorodriguez
 */
public class TituloPropiedad {
    
    private String nombre;
    private boolean hipotecada=false;
    private int alquilerBase;
    private float factorRevalorizacion;
    private int hipotecaBase;
    private int precioEdificar;
    
    //Atributo de la clase CASILLA
    private Casilla casilla;
    
    //Atributo de la clase JUGADOR
    private Jugador propietario;
    
    
    //-----------CONSTRUCTOR POR PARAMETROS------------
    //PROPIETARIO AL CREARLO ES NULO
    public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar ){
        this.nombre=nombre;
        this.alquilerBase=alquilerBase;
        this.factorRevalorizacion=factorRevalorizacion;
        this.hipotecaBase=hipotecaBase;
        this.precioEdificar=precioEdificar;   
    }
    
    //  CONSTRUCTOR PARA VER SI TIENE PROPIETARIO
    public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar, Jugador propiertario, Casilla casilla ){
        this.nombre=nombre;
        this.alquilerBase=alquilerBase;
        this.factorRevalorizacion=factorRevalorizacion;
        this.hipotecaBase=hipotecaBase;
        this.precioEdificar=precioEdificar;
        this.propietario = propietario;
        this.casilla = casilla;
    }

     
    // ---------------METODOS GETS - SETS----------------
    //VISIBILIDAD CAMBIADA A PACKAGE EN LA P2 EN LOS METODOS GET
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    boolean getHipotecada() {
        return hipotecada;
    }

    public void setHipotecada(boolean hipotecada) {
        this.hipotecada = hipotecada;
    }

    int getAlquilerBase() {
        return alquilerBase;
    }
    
    public void setAlquilerBase(int alquilerBase) {
        this.alquilerBase = alquilerBase;
    }

    float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }

    public void setFactorRevalorizacion(float factorRevalorizacion) {
        this.factorRevalorizacion = factorRevalorizacion;
    }

    int getHipotecaBase() {
        return hipotecaBase;
    }

    public void setHipotecaBase(int hipotecaBase) {
        this.hipotecaBase = hipotecaBase;
    }

    int getPrecioEdificar() {
        return precioEdificar;
    }

    public void setPrecioEdificar(int precioEdificar) {
        this.precioEdificar = precioEdificar;
    }
    
    //METODOS NUEVOS P2
    Jugador getPropietario(){
        return propietario;
    }
    
    void setPropietario(Jugador propietario){
        this.propietario = propietario;
    }
    
    Casilla getCasilla(){
        return casilla;
    }
    
    void setCasilla(Casilla casilla){
        this.casilla = casilla;
    }
    
    void cobrarAlquiler(int costeAlquiler){
        propietario.modificarSaldo(costeAlquiler);
    }
    
    boolean propietarioEncarcelado(){
        return propietario.getEncarcelado();
    }
    
    boolean tengoPropietario(){
        return (propietario != null);
    }
    
    
    @Override
    public String toString() {
//        if(tengoPropietario() == true)
            return "{" + "Nombre: " + nombre + ", Hipotecada: " + hipotecada + ", Alquiler Base: " + alquilerBase + ", Factor Revalorizacion=" + factorRevalorizacion + ", Hipoteca Base: " + hipotecaBase + ", Precio Edificar: " + precioEdificar + '}'; //"Propietario: " + propietario.toString() + "Casilla: " + casilla.toString() +'}';
       // else
         //   return "{" + "nombre=" + nombre + ", hipotecada=" + hipotecada + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + '}';
   
    }
    
    
    
    
}
